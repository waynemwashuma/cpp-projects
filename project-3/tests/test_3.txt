a, b, c, d, x, y, z;

main {
    public:
        a, b, c;
    private:
        d, e, f, g;

    d = a;
    b = c;
    sub1 {
        public:
            a, x, w;

        a = b;
        d = a;
        e = c;
    }
    x = g;
    f = w;
    sub2 {
        public:
            b;
        private:
            y, z;

        myScope {
            y = a;
            b = z;
        }
        y = a;
        b = z;
    }
    y = a;
    b = z;
}