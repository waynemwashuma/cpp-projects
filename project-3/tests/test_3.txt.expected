main.d = main.a
main.b = main.c
sub1.a = main.b
::d = sub1.a
?.e = main.c
::x = main.g
main.f = ?.w
::y = main.a
sub2.b = ::z
sub2.y = main.a
sub2.b = sub2.z
::y = main.a
main.b = ::z
